script.js

```diff
@@ -1,12 +1,265 @@
+    console.log("You clicked option " + event.target.value);
+    let urlOfApi = `${shows[event.target.value]._links.self.href}/episodes`;
+    getEpisodesFromSelectedShow(urlOfApi);
+  });
+}
+
+async function getEpisodesFromSelectedShow(url) {
+  //this function will fetch episodes based on the selected show
+  let episodesFromShow = await fetchFromApi1(url);
+  console.log(episodesFromShow);
+  makePageForEpisodes(episodesFromShow);
+  selectEpisode(episodesFromShow);
+  makeLiveSearch(episodesFromShow);
+}
+
+//make it for shows then I'll try to make it general
+async function fetchFromApi1(url) {
+  try {
+    const response = await fetch(url);
+
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error fetching data:", error);
+  }
+}
+
+function selectEpisode(episodes) {
+  let headerEl = document.getElementById("header-id");
+  //trying to avoid duplication in episodes by keeping show div and creating everything else here
+  let selectShowDivEl = document.getElementById("select-show-el");
+  headerEl.innerHTML = ""; //its not working as its already there
+  headerEl.appendChild(selectShowDivEl);
+
+  let selectEpisodeDivEl = document.createElement("div");
+  selectEpisodeDivEl.innerHTML = "";
+  selectEpisodeDivEl.id = "select-div-el";
+  headerEl.appendChild(selectEpisodeDivEl);
+  let selectTitleEl = document.createElement("h4");
+  selectTitleEl.textContent = "Select Episode...";
+  selectEpisodeDivEl.appendChild(selectTitleEl);
+  let selectEl = document.createElement("select");
+  selectEpisodeDivEl.appendChild(selectEl);
+
+  for (let i = 0; i < episodes.length; i++) {
+    let option = document.createElement("option");
+    option.value = i;
+    option.text = `${episodes[i].name} - S0${episodes[i].season}E0${episodes[i].number}`;
+    selectEl.appendChild(option);
+  }
+
+  selectEl.addEventListener("click", function (event) {
+    console.log("You clicked option " + event.target.value);
+    let arrayAfterEpisodeSelect = [episodes[event.target.value]];
+    makePageForEpisodes(arrayAfterEpisodeSelect);
+  });
+}
+
+function makeLiveSearch(allEpisodes) {
+  let searchTitleEl = document.createElement("h4");
+  searchTitleEl.textContent = "Search....";
+  let searchBoxEl = document.createElement("input");
+  searchBoxEl.placeholder = "Search here ......";
+  let headerEl = document.getElementById("header-id");
+  headerEl.appendChild(searchTitleEl);
+  headerEl.appendChild(searchBoxEl);
+  let displayFilteredEpisodeNumberP = document.createElement("p");
+  headerEl.appendChild(displayFilteredEpisodeNumberP);
+
+  searchBoxEl.addEventListener("input", function () {
+    let searchedWords = searchBoxEl.value;
+    let episodesAfterSearch = isEpisodeIncludeWord(allEpisodes, searchedWords);
+
+    //to display number of results from the search
+    let lengthOfFilteredEpisodes =
+      episodesAfterSearch.length > 0 ? episodesAfterSearch.length : 0;
```

Comment:
Here we could just return `episodesAfterSearch.length` since if it has nothing inside, it will be 0 anyway

Reply:
Brilliant !!